naresh/src/main/java/com/lftechnology/phpjava/ems/views/EmployeeView.java

```diff
@@ -0,0 +1,373 @@
+package com.lftechnology.phpjava.ems.views;
+
+import com.lftechnology.phpjava.ems.constants.Constant;
+import com.lftechnology.phpjava.ems.controllers.EmployeeController;
+import com.lftechnology.phpjava.ems.entities.Employee;
+import com.lftechnology.phpjava.ems.enums.Role;
+import com.lftechnology.phpjava.ems.utlis.CommonUtility;
+import com.lftechnology.phpjava.ems.utlis.ConsoleWriter;
+import com.lftechnology.phpjava.ems.utlis.Router;
+import com.lftechnology.phpjava.ems.utlis.UserInput;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * EmployeeView
+ *
+ * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+ * @since August, 16 2016
+ */
+public class EmployeeView implements ViewSignature<Object> {
+    public static CommonUtility commonUtility = new CommonUtility();
+    private EmployeeController employeeController = new EmployeeController();
+    private String action;
+
+    /**
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    @Override
+    public void render() {
+        switch (action) {
+            case Constant.POST_LOGIN_ACTION:
+                if (commonUtility.isLoggedIn() && commonUtility.isAdmin()) {
+                    this.showAdminMenu();
+                } else {
+                    this.showNormalUserMenu();
+                }
+                break;
+            case Constant.ADD_USER_ACTION:
+                this.showAddUserMenu();
+                break;
+            case Constant.DELETE_USER_ACTION:
+                this.showDeleteUserMenu();
+                break;
+            case Constant.SEARCH_USER_ACTION:
+                this.showSearchUserMenu();
+                break;
+            case Constant.TERMINATE_USER_ACTION:
+                this.showTerminateUserMenu();
+                break;
+            case Constant.UPDATE_PROFILE_ACTION:
+                this.showUpdateUserMenu();
+                break;
+            case Constant.EXIT_ACTION:
+                CommonViewUtility.showExitMessageAndExit();
+                break;
+            default:
+                Router.showMenu();
+        }
+    }
+
+    /**
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    @Override
+    public String getAction() {
+        return this.action;
+    }
+
+    /**
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    @Override
+    public void setAction(String action) {
+        this.action = action;
+    }
+
+    /**
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    @Override
+    public Map<String, Object> getData() {
+        return null;
+    }
+
+    /**
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    @Override
+    public void setData(Map<String, Object> data) {
+        commonUtility = (CommonUtility) data.get(Constant.COMMON_UTILITY);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Make the enclosing method "static" or remove this set. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.java.lftechnology.com:9000/coding_rules#rule_key=squid%3AS2696)